### PR TITLE
docs: refactor access control configuration sections

### PR DESCRIPTION
**Pull request by james-d-elliott**
Refactors the access-control configuration documentation to be up-to-date and conform to our style guidelines. Additionally went over each part and reworded things that needed it.

TODO:
- [x] Add styled labels.

### CODE REVIEW

**Review by clems4ever — DISMISSED** [review]
Tiny remarks otherwise LGTM.

File: docs/configuration/access-control.md [hunk 1]
```diff
@@ -8,176 +8,341 @@ nav_order: 1
...
+* [subject](#subject): the user or group of users to define the policy for.
+* [networks](#networks): the network addresses, ranges (CIDR notation) or groups from where the request originates.
+* [methods](#methods): the http methods used in the request.
 
-## Rules
+A rule is matched when all criteria of the rule match. Rules are evaluated in sequential order, and this is
+particularly **important** for bypass rules. Bypass rules should generally appear near the top of the rules list.
+However you need to carefully evaluate your rule list **in order** to see which rule matches a particular scenario. A
+comprehensive understanding of how rules apply is also recommended. ***Note:** we could theoretically devise a tool that
+policy output given input of a users request and a rule list in the future.*
```
[the remark this answers is not in the document]
Comment (james-d-elliott) [reply docs/configuration/access-control.md:87]: Yeah that's a good idea.

File: docs/configuration/access-control.md [hunk 2]
```diff
@@ -8,176 +8,341 @@ nav_order: 1
...
 
-### Networks
+This criteria matches the HTTP request method. This is primarily useful when trying to bypass authentication for specific
+request types when those requests would prevent essential or public operation of the website. An example is when you
+need to do CORS preflight requests you could apply the `bypass` policy to `OPTIONS` requests.
 
-A list of network addresses, ranges (CIDR notation) or groups can be specified in a rule in order to apply different
-policies when requests originate from different networks. This list can contain both literal definitions of networks
-and [network aliases](#network-aliases).
+It's important to note that Authelia cannot preserve request data when redirecting the user. For example if the user had
```
[the remark this answers is not in the document]
Comment (james-d-elliott) [reply docs/configuration/access-control.md:214]: Thanks, felt like it was essential we communicate this clearly.

File: docs/configuration/access-control.md [hunk 3]
```diff
@@ -8,176 +8,395 @@ nav_order: 1
...
+complicated network related configuration a lot cleaner and easier to read.
 
-This policy requires the user to complete 2FA successfully. This is currently the highest level of authentication
-policy available.
+This section has two options, `name` and `networks`. Where the `networks` section is a list of IP addresses in CIDR
+notation and where `name` is a friendly name to label the collection of networks for reuse in the [networks](#networks) 
+section of the [rules](#rules) section below.
 
-## Default Policy
+This configuration option *does nothing* by itself, it's only useful if you use theese aliases in the [rules](#networks)
```
Comment (nightah) [inline docs/configuration/access-control.md:75]:
```suggestion
This configuration option *does nothing* by itself, it's only useful if you use these aliases in the [rules](#networks)
```

File: docs/configuration/access-control.md [hunk 4]
```diff
@@ -8,176 +8,395 @@ nav_order: 1
...
+### networks
+<div markdown="1">
+type: list(string)
+{: .label .label-config .label-purple } 
+required: no
+{: .label .label-config .label-green }
+</div>
 
-### Resources
+This criteria is a list of network address ranges in CIDR notation or an alias from the [global](#networks-global)
```
Comment (nightah) [inline docs/configuration/access-control.md:284]: The network addresses do not need to be in CIDR notation, they can be singlular IP addresses or CIDR notation.

**Review by nightah — CHANGES_REQUESTED** [review]
Two tiny nits, otherwise LGTM.

**Review by nightah — APPROVED** [review]

**Pull request merged**